gravitee-node-kubernetes/src/main/java/io/gravitee/node/kubernetes/keystoreloader/KubernetesConfigMapKeyStoreLoader.java

```diff
@@ -117,41 +124,82 @@ protected Flowable<ConfigMap> watch() {
 
     @Override
     protected Completable loadKeyStore(ConfigMap configMap) {
-        final Optional<ResourceQuery<ConfigMap>> optResource = resources
-            .values()
-            .stream()
-            .filter(r ->
-                r.getNamespace().equalsIgnoreCase(configMap.getMetadata().getNamespace()) &&
-                r.getResource().equalsIgnoreCase(configMap.getMetadata().getName())
-            )
-            .findFirst();
-
-        if (optResource.isEmpty()) {
-            return Completable.error(new IllegalArgumentException("Unable to load keystore: unknown configmap."));
+        KeyStore keyStore = null;
```

Comment:
Looks like this method becomes a bit large. Can you try to improve the readability(e.g. create private methods for each case)?